FE/src/hooks/useShellQuery.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  useQuery,
+  useMutation,
+  UseMutationOptions,
+  useQueryClient,
+} from 'react-query'
+import { fetchShells, addShell, deleteShell } from '@/api/shellApi'
+import { ShellType } from '@/types/interfaces'
+
+const QUERY_KEYS = {
+  shells: 'shells',
+} as const
+
+type MutationContext = {
+  previousShells: ShellType[] | undefined
+}
+
+function useShellMutation<TData, TVariables>(
+  mutationFn: (variables: TVariables) => Promise<TData>,
+  options?: UseMutationOptions<TData, Error, TVariables, MutationContext>
+) {
+  const queryClient = useQueryClient()
+
+  return useMutation<TData, Error, TVariables, MutationContext>(mutationFn, {
+    onMutate: (newShell) => {
+      const previousShells = queryClient.getQueryData(QUERY_KEYS.shells)
+
+      queryClient.setQueryData<ShellType[]>(QUERY_KEYS.shells, (old) =>
+        old ? [...old, newShell as ShellType] : [newShell as ShellType]
```

Comment:
여기서 old가 배열로 받으면 항상  [...old, newShell as ShellType] 만 실행되고 [newShell as ShellType] 부분은 실행되지 않을 것 같아요.
```suggestion
      queryClient.setQueryData<ShellType[]>(QUERY_KEYS.shells, (old = []) =>
        [...old, newShell as ShellType]
```
이렇게 바꿔줘도 괜찮지 않을까요?

Reply:
넵 해당 부분 수정하겠습니다.